src/System.IO.Pipelines.Text.Primitives/ReadableBufferExtensions.cs

```diff
@@ -210,7 +210,11 @@ public static string GetUtf8String(this ReadableBuffer buffer)
                 return null;
             }
 
-            ReadOnlySpan<byte> textSpan = stackalloc byte[0];
+            // TODO: "ReadOnlySpan<byte> textSpan = stackalloc byte[0]" would fit better here, 
+            //       but it emits substandard IL, see https://github.com/dotnet/roslyn/issues/21952
+            //
+            // make 'textSpan' formally stack-referring or we won't be able to reference stack data later
```

Comment:
I know you didn't write the comment @VSadov (copy/pasted), but I don't understand it.

What does it mean by 'formally stack-referring'. Also, shouldn't it read:
make 'textSpan' formally stack-referring or **we will be able to** reference stack data later 

If textSpan is a stackalloc variable, we can't reference it outside the method, correct? The comment seems to suggest that if we don't make it 'stack-referring', we won't be able to reference it later, which has the opposite meaning.

Reply:
If we do not assign anything, 'textSpan' will default to be a returnable variable. We will be able to return it from the method, but will not be able to assign stackalloc'd spans. That is what you normally want, but not in this case. Here we need to initialize to something stackallocated to intentionally "taint" the variable as "possibly referring to stack data". 

'stackalloc byte[0]' would be a nice choice since it is basically the same as 'default(Span<T>)', but formally stack-allocated. However we do a poor job with optimizing such trivial case, so I am using a more verbose, but reliable, way to essentially assign a default span, while forcing the variable be classified as stack-referring.

I will try to change the comment to be more clear.